packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -209,6 +214,42 @@ export default createRule<Options, MessageIds>({
               );
             }
           });
+          return;
+        }
+
+        if (node.typeAnnotation) {
```

Comment:
Yea, you are right! It would be 10x better if we relied on the actual type rather than the syntax around the destructuring. That way we can actually handle `VariableDeclarator`s without `init`:

```ts
class Unbound {
  method() { }
}

declare const foo: Unbound[]
for (const { method } of foo) {}
```